lib/reversetunnel/localsite.go

```diff
@@ -262,6 +242,47 @@ func (s *localSite) dialWithAgent(params DialParams) (net.Conn, error) {
 	return conn, nil
 }
 
+// dialTunnel connects to the target host through a tunnel.
+func (s *localSite) dialTunnel(params DialParams) (net.Conn, error) {
+	rconn, err := s.getRemoteConn(params.ServerID)
+	if err != nil {
+		return nil, trace.NotFound("no tunnel connection found: %v", err)
+	}
+
+	s.log.Debugf("Tunnel dialing to %v.", params.ServerID)
+
+	conn, err := s.chanTransportConn(rconn)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return conn, nil
+}
+
+func (s *localSite) getConn(params DialParams) (net.Conn, bool, error) {
```

Comment:
Nit: Named return arguments would really help in this case, e.g. to understand what the bool parameter is.